stablehlo/dialect/VhloOps.td

```diff
@@ -65,246 +65,246 @@ class VHLO_Op<string mnemonic, string minVersion = "0.3.0", string maxVersion =
+  let results = (outs VHLO_AnyType:$result);
 }
 def VHLO_PowOpV1 : VHLO_Op<"power"> {
-  let arguments = (ins VHLO_Tensor:$lhs, VHLO_Tensor:$rhs);
-  let results = (outs VHLO_Tensor:$result);
+  let arguments = (ins VHLO_AnyType:$lhs, VHLO_AnyType:$rhs);
+  let results = (outs VHLO_AnyType:$result);
 }
 def VHLO_RemOpV1 : VHLO_Op<"remainder"> {
-  let arguments = (ins VHLO_Tensor:$lhs, VHLO_Tensor:$rhs);
-  let results = (outs VHLO_Tensor:$result);
+  let arguments = (ins VHLO_AnyType:$lhs, VHLO_AnyType:$rhs);
+  let results = (outs VHLO_AnyType:$result);
 }
 def VHLO_ShiftLeftOpV1 : VHLO_Op<"shift_left"> {
-  let arguments = (ins VHLO_Tensor:$lhs, VHLO_Tensor:$rhs);
-  let results = (outs VHLO_Tensor:$result);
+  let arguments = (ins VHLO_AnyType:$lhs, VHLO_AnyType:$rhs);
+  let results = (outs VHLO_AnyType:$result);
 }
 def VHLO_ShiftRightArithmeticOpV1 : VHLO_Op<"shift_right_arithmetic"> {
-  let arguments = (ins VHLO_Tensor:$lhs, VHLO_Tensor:$rhs);
-  let results = (outs VHLO_Tensor:$result);
+  let arguments = (ins VHLO_AnyType:$lhs, VHLO_AnyType:$rhs);
+  let results = (outs VHLO_AnyType:$result);
 }
 def VHLO_ShiftRightLogicalOpV1 : VHLO_Op<"shift_right_logical"> {
-  let arguments = (ins VHLO_Tensor:$lhs, VHLO_Tensor:$rhs);
-  let results = (outs VHLO_Tensor:$result);
+  let arguments = (ins VHLO_AnyType:$lhs, VHLO_AnyType:$rhs);
+  let results = (outs VHLO_AnyType:$result);
 }
 def VHLO_SubtractOpV1 : VHLO_Op<"subtract"> {
-  let arguments = (ins VHLO_Tensor:$lhs, VHLO_Tensor:$rhs);
-  let results = (outs VHLO_Tensor:$result);
+  let arguments = (ins VHLO_AnyType:$lhs, VHLO_AnyType:$rhs);
+  let results = (outs VHLO_AnyType:$result);
 }
 
 // Logical Ops
 def VHLO_AndOpV1 : VHLO_Op<"and"> {
-  let arguments = (ins VHLO_PredOrIntTensor:$lhs, VHLO_PredOrIntTensor:$rhs);
-  let results = (outs VHLO_Tensor:$result);
+  let arguments = (ins VHLO_AnyType:$lhs, VHLO_AnyType:$rhs);
+  let results = (outs VHLO_AnyType:$result);
 }
 def VHLO_OrOpV1 : VHLO_Op<"or"> {
-  let arguments = (ins VHLO_PredOrIntTensor:$lhs, VHLO_PredOrIntTensor:$rhs);
-  let results = (outs VHLO_Tensor:$result);
+  let arguments = (ins VHLO_AnyType:$lhs, VHLO_AnyType:$rhs);
+  let results = (outs VHLO_AnyType:$result);
 }
 def VHLO_XorOpV1 : VHLO_Op<"xor"> {
-  let arguments = (ins VHLO_PredOrIntTensor:$lhs, VHLO_PredOrIntTensor:$rhs);
-  let results = (outs VHLO_Tensor:$result);
+  let arguments = (ins VHLO_AnyType:$lhs, VHLO_AnyType:$rhs);
+  let results = (outs VHLO_AnyType:$result);
 }
 
 def VHLO_ReturnOpV1 : VHLO_Op<"return", "0.3.0", "current", [Terminator]> {
   let arguments = (ins
-    Variadic<VHLO_TensorOrTokenOrTuple >:$results
+    Variadic<VHLO_AnyType>:$results
   );
   let assemblyFormat = "$results attr-dict (`:` type($results)^)?";
 }
 
 // Communication op definitions.
 def VHLO_InfeedOpV1 : VHLO_Op<"infeed"> {
   let arguments = (ins
-    VHLO_Token:$token,
+    VHLO_AnyType:$token,
     DefaultValuedStrAttr<StrAttr, "">:$infeed_config,
```

Comment:
Can we change default valued attrs to `OptionalAttr<VHLO_AnyAttr>`? (Here and below).

Reply:
Should it be optional or just `VHLO_AnyAttr`? Default valued means it will always have a value?

Reply:
That's exactly what I had in mind. Will make those changes